### PR TITLE
Adding mvapich support

### PR DESCRIPTION
**Pull request by j34ni**
Checklist
* [X] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

Here I tried to add MVAPIch support (latest version, regardless of the netmod, for now), keeping version 1.14.4 instead of moving to 1.14.5.

### CODE REVIEW

**Comment by j34ni** [comment]
@conda-forge-admin, please rerender

**Comment by hmaarrfk** [comment]
ping again if/when ready or you need help.

**Comment by j34ni** [comment]
@hmaarrfk Thanks, I will get in touch very soon then

**Comment by hmaarrfk** [comment]
unfortunately i do'nt have time. please see the other PRs where others have been working on something similar.

**Comment by j34ni** [comment]
@hmaarrfk Is it all good now?

File: recipe/dummy_t_filters_parallel.c [hunk 1]
```diff
@@ -0,0 +1,6 @@
+#include <stdio.h>
+int main(void) {
+    printf("Replacement test for t_filters_parallel\n");
```
Comment (hmaarrfk) [inline recipe/dummy_t_filters_parallel.c:3]: What do these test files do?

Comment (j34ni) [reply recipe/dummy_t_filters_parallel.c:3]: @hmaarrfk The `t_filters_parallel` test suite ensures the correct application and integrity of HDF5 filters, such as compression, in a parallel I/O context. The `t_pmulti_dset` test suite verifies the proper creation and I/O operations on multiple datasets in parallel. We had to disable these tests for MVAPICH due to specific failures for a couple of them, likely related to resource constraints in the testing environment.

Comment (j34ni) [reply recipe/dummy_t_filters_parallel.c:3]: Is there anything else needed from my side before this PR can be merged?

Comment (hmaarrfk) [reply recipe/dummy_t_filters_parallel.c:3]: Can you add this statement as a comment in the source

Comment (j34ni) [reply recipe/dummy_t_filters_parallel.c:3]: Done

File: recipe/build.sh [hunk 2]
```diff
@@ -88,7 +85,7 @@ if [[ "$CONDA_BUILD_CROSS_COMPILATION" == 1 && $target_platform == "osx-arm64" ]
   # Do we need this below?
   export hdf5_cv_szlib_can_encode=yes
 
-  HDF5_OPTIONS="${HDF5_OPTIONS} --enable-tests=no"
+  HDF5_OPTIONS="${HDF5_OPTIONS} --enable-tests=no "
```
Comment (hmaarrfk) [inline recipe/build.sh:88]: is this additional space important?

Comment (hmaarrfk) [reply recipe/build.sh:88]:
```suggestion
  HDF5_OPTIONS="${HDF5_OPTIONS} --enable-tests=no"
```

**Comment by hmaarrfk** [comment]
@conda-forge-admin please rerender

Please remember to bump the build number next time too!

Sorry i had to review how you "fixed" osx-arm. it seems you reverted to the older version

**Comment by hmaarrfk** [comment]
@conda-forge-admin please restart cis

**Comment by hmaarrfk** [comment]
why did it magically start to fail?

**Comment by hmaarrfk** [comment]
you may unfortunately have to rebase: https://github.com/conda-forge/hdf5-feedstock/pull/239

**Comment by hmaarrfk** [comment]
i mostly just wanted to trigger a build

**Comment by j34ni** [comment]
@conda-forge-admin, please rerender

**Comment by hmaarrfk** [comment]
hint: you must bump the build number again

**Comment by j34ni** [comment]
@hmaarrfk It seems that the build for the HDF5 package on Linux with MVAPICH was failing due to missing symbols related to `libacl` (which for some reason was not needed until now?!), and adding this package on the host resolved the issue, allowing the build to proceed successfully.
However there is now an issue with win64 and '".scripts\create_conda_build_artifacts.bat"' is not recognized as an internal or external command: did I mess up something else?

**Comment by hmaarrfk** [comment]
rebasing is etter than merging.

when you rebase, drop all the MNT commits from rerendering

**Comment by j34ni** [comment]
@conda-forge-admin, please rerender

**Comment by j34ni** [comment]
@hmaarrfk `libacl` is not required at runtime, only during the build, so I guess we can add;
```
  ignore_run_exports:
    - libacl
```

**Comment by hmaarrfk** [comment]
> @hmaarrfk `libacl` is not required at runtime, only during the build, so I guess we can add;
> 
> ```
>   ignore_run_exports:
>     - libacl
> ```

yes.

**Comment by hmaarrfk** [comment]
please add a comment referring to this PR.

**Comment by j34ni** [comment]
Or should I simply add it myself (I did not want to interrupt the build on Azure, which is now becoming all green)?

**Comment by hmaarrfk** [comment]
choose to interupt at your leisure or to let pass, but given this discussion, i'll wait until you  you add the ignore run export and comment before we merge into main

**Comment by hmaarrfk** [comment]
if you have the git skills, it would be good to rebase this into a single commit + and a single rerender

**Comment by j34ni** [comment]
@conda-forge-admin, please rerender

**Comment by j34ni** [comment]
Thanks @hmaarrfk 
Now I just have to continue with NetCDF, etc.

**Pull request merged**